mobx/test/observable_list_test.dart

```diff
@@ -98,20 +99,72 @@ void main() {
       expect(count, equals(2));
     });
 
+    test('observe set length works', () {
+      final list = ObservableList.of([0]);
+
+      var changesCount = 0;
+      var index = -1;
+      var addedValues = <int>[];
+      var removedValues = <int>[];
+
+      list.observe((change) {
+          changesCount += change.rangeChanges?.length ?? 0;
```

Comment:
Can you simplify these tests for all the newly added tests. I think there is excessive conditional logic inside the tests. We can instead make explicit checks for the **before - after** changes, rather than applying conditional logic. That will improve the readability as well maintainability.